File: LeetCode/LeetCode Practice Problems/1630- Arithmetic Subarrays.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        vector<bool> ans;
        
        int sz = l.size();
        for(int i=0;i<sz;i++) {
            int start = l[i];
            int end = r[i];
            
            vector<int> v;
            for(int j=start;j<=end;j++) {
                v.push_back(nums[j]);
            }
            
            sort(v.begin() , v.end());
            bool f = true;
            
            for(int j=2;j<v.size();j++) {
                if(v[j]-v[j-1] != v[1] - v[0]) {
                    f = false;
                    break;
                }
            }
            
            ans.push_back(f);
        }
        
        
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/1630-Arithmetic Subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> ans = new ArrayList();
        
        for(int i=0;i<l.length;i++) {
            int sz = r[i] - l[i] + 1;
            int a[] = new int[sz];
            
            for(int j=l[i];j<=r[i];j++) {
                a[j - l[i]] = nums[j];
            }
            
            Arrays.sort(a);
            int val = a[1] - a[0];
            boolean f = true;
            
            for(int j=2;j<sz;j++) {
                if(a[j]-a[j-1] != val) {
                    f = false;
                    break;
                }
            }
            
            ans.add(f);
        }
        
        return ans;
    }
}
